pull_request_template.md

```diff
@@ -0,0 +1,20 @@
+<!--- Please fill your JIRA ticket in below, both in the text and link. -->
+Relevant JIRA ticket: [KZN-XXX](https://kzngroup.atlassian.net/browse/KZN-XXX)
+
+## Description of Changes
+
+## Change type
+
+<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+
+## Checklist
+
+<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
+<!--- If one of the below isn't complete, please go and polish that off before creating a PR. -->
+- [ ] Linting has been applied to my code.
+- [ ] Relevant documentation has been updated.
+- [ ] Tests have been written where possible.
+- [ ] I have satisfied the [KZN Definition of Done](https://kzngroup.atlassian.net/wiki/spaces/COMMUNITY/pages/209059874/KZN+Definition+of+Done).
```

Comment:
I like the introduction of an org wide template.

I don't think the 'change type' heading is useful, we don't really do semver and don't distinguish between bug/feature, so I don't think that information impacts anything.

On the checklist, I think either it should only link to DoD or it should list the DoD items.

Things we might want to consider adding:
- A box indicating whether this is an early PR or we think it's complete.
- A prompt to link doco.

Extreme opinion: Include verbiage that by clicking 'approve' you accept maintaining this code.

Reply:
Fair point on the change type. I've used that at previous jobs but mainly because there is a different testing process for the type of change we're making- which you're right, we don't have here. Happy to remove if thats the agreement.
Added an explicit link for documentation already.

Reply:
can the check-boxes automatically apply labels? as that sounds like what I might use to indicate early/WIP feedback

Reply:
yeah we can probably continue to use the prefix in branch names to indicate feature vs hotfix, these are probably more likely to be used by code pipelines to determine special logic or approval requirements already